nengo/synapses.py

```diff
@@ -161,41 +244,50 @@ class LinearFilter(Synapse):
     .. [1] https://en.wikipedia.org/wiki/Filter_%28signal_processing%29
     """
 
-    num = NdarrayParam("num", shape="*")
-    den = NdarrayParam("den", shape="*")
-    analog = BoolParam("analog")
-    method = EnumParam(
-        "method", values=("gbt", "bilinear", "euler", "backward_diff", "zoh")
-    )
+    _argrepr_filter = {"den"}
 
-    def __init__(self, num, den, analog=True, method="zoh", **kwargs):
-        super().__init__(**kwargs)
-        self.num = num
-        self.den = den
-        self.analog = analog
-        self.method = method
-
-    def combine(self, obj):
-        """Combine in series with another LinearFilter."""
-        if not isinstance(obj, LinearFilter):
+    def __init__(
+        self,
+        sys,
+        den=None,
+        analog=True,
+        method="zoh",
+        x0=0,
+        initial_output=None,
+        **kwargs
+    ):
+        if den is not None:
+            warnings.warn(
+                DeprecationWarning(
+                    "`den` is deprecated. Pass systems in transfer function form as a "
+                    "`(numerator, denominator)` 2-tuple instead."
+                )
+            )
+            sys = (sys, den)
```

Comment:
`num` and `den` are still listed as attributes of this class, but I don't think they are anymore. Maybe they should be attributes of `LinearSystem`?

Reply:
I removed them. They're not attributes of `LinearSystem` either, since it stores things in state-space form. It does have the attributes `A`, `B`, `C`, and `D`, but I haven't listed them there because the same information can be accessed through the `ss` property, which *is* documented.